eslint-plugin-kuc-v1/super-update.js

```diff
@@ -0,0 +1,37 @@
+module.exports = {
+  create: function(context) {
+    return {
+      ClassDeclaration: function(node) {
+        const superClass = node.superClass.name;
+        if (superClass !== "KucBase") return;
+
+        const nodeBody = node.body;
+        if (nodeBody.type !== "ClassBody") return;
+
+        const sourceCode = context.getSourceCode();
+        const pattern = "super\\.update\\(";
+        const regex = new RegExp(pattern);
+
+        nodeBody.body.forEach(body => {
+          if (
+            body.type === "MethodDefinition" &&
+            body.key.type === "Identifier" &&
+            body.key.name === "update" &&
+            !body.computed
+          ) {
+            const functionSourceCode = sourceCode.getText(body.value);
+            if (regex.test(functionSourceCode)) return;
+
+            context.report({
+              node: body,
+              message: `Please write super.update(changedProperties) using PropertyValues to map the changed properties with old values.
+              ex: update(changedProperties: PropertyValues) {
+                    super.update(changedProperties);
+                  }`
```

Comment:
This is the rule described here.
https://sharedoc.atlassian.net/wiki/spaces/SC/pages/906166341/EN+KUC+v1.x.x+Develop+env+coding+rule#Error-Handling

How about updating the explanation to be more clear.
```
message: `Please write super.update(changedProperties) using PropertyValues to map the changed properties with old values when using validator.
ex: update(changedProperties: PropertyValues) {
      if (changedProperties.has("items")) {
        validateItems(this.items);
      }
      super.update(changedProperties);
    }`
```

Reply:
I see that there are several lint errors regarding this in component scripts (base component, button, etc), but the error won't be displayed when executing `npm run es-lint`.
Could you please check it?

Reply:
\>>How about updating the explanation to be more clear.
Writing super.update() in update() is not only for validator, but also for updating the element’s attributes and template.
Ref. the "update()" of https://lit.dev/docs/components/lifecycle/#reactive-update-cycle-performing
So I think we do not need to add "when using validator".

\>>I see that there are several lint errors regarding this in component scripts (base component, button, etc)
Is it the error like following?
ESLint: Definition for rule 'kuc-v1/super-update' was not found.(kuc-v1/super-update)
If YES, please run ```npm ci``` again.

Reply:
> Writing super.update() in update() is not only for validator, but also for updating the element’s attributes and template.
> So I think we do not need to add "when using validator".

I see. Your suggestion seems correct.
Please use your previous sentence.

> Is it the error like following?
> ESLint: Definition for rule 'kuc-v1/super-update' was not found.(kuc-v1/super-update)
> If YES, please run npm ci again.

The answer is YES, and as you suggested the error is disappeared after installing packages again.
Thank you for following up!